src/js/models/menuitem.js

```diff
@@ -283,16 +320,22 @@ define(function (require, exports, module) {
      * @return {MenuItem}
      */
     MenuItem.prototype.updateSubmenuProps = function (submenuID, props) {
-        var menuItem = this.submenuMap.get(submenuID),
-            menuIndex = this.submenu.indexOf(menuItem);
+        var menuIndex = this.submenu.findIndex(function (menuItem) {
```

Comment:
I was trying to avoid building it unnecessarily, but... now that I say that out loud, it doesn't really make any sense. Will fix.